src/Dialog/Dialog.js

```diff
@@ -121,7 +125,7 @@ function getStyles(props, context) {
       WebkitTapHighlightColor: 'rgba(0,0,0,0)', // Remove mobile color flashing (deprecated)
       padding: 8,
       width: '100%',
-      textAlign: 'right',
+      textAlign: isRtl ? 'left' : 'right',
```

Comment:
My be we should remove that logic instead https://github.com/callemall/material-ui/blob/master/src/utils/rtl.js#L40 ?